examples/seismic/source.py

```diff
@@ -1,60 +1,106 @@
+from scipy import signal
+
 from devito import Dimension
 from devito.function import SparseTimeFunction
 from devito.logger import error
 
+from cached_property import cached_property
+
 import numpy as np
 try:
     import matplotlib.pyplot as plt
 except:
     plt = None
 
-__all__ = ['PointSource', 'Receiver', 'Shot', 'RickerSource', 'GaborSource']
+__all__ = ['PointSource', 'Receiver', 'Shot', 'WaveletSource',
+           'RickerSource', 'GaborSource', 'TimeTicks']
+
+
+class TimeTicks:
+    def __init__(self, start=None, step=None, num=None):
+        try:
+            self.start = start
+            self.stop = start + step*num
+            self.step = step
+            self.num = num
+        except:
+            error("error in time range parameters")
```

Comment:
Wouldn't it be simpler to just make input non keyword to fore it as input?

Reply:
Agreed. Committed b99883be

Reply:
why would you get an exception here? and we should do more than an error message (perhaps by raising a ``ValueError`` exception?